addons/mrp/models/mrp_production.py

```diff
@@ -478,12 +490,10 @@ def onchange_picking_type(self):
 
     def write(self, vals):
         res = super(MrpProduction, self).write(vals)
-        if 'date_planned_start' in vals:
-            moves = (self.mapped('move_raw_ids') + self.mapped('move_finished_ids')).filtered(
-                lambda r: r.state not in ['done', 'cancel'])
-            moves.write({
-                'date_expected': fields.Datetime.to_datetime(vals['date_planned_start']),
-            })
+
+        if not self.routing_id and vals.get('date_planned_start') and 'date_planned_finished' not in vals:
+            self.date_planned_finished = self.date_planned_start + datetime.timedelta(hours=1)
```

Comment:
you could do it before the call to super and only set the vals (no write inside write)